File: PostBox/src/main/java/com/mvp/bean/BaseBox.java
package com.mvp.bean;

public abstract class BaseBox {
    private float length;
    private float width;
    private float height;

    protected BaseBox(float length, float width, float height) {
        this.length = length;
        this.width = width;
        this.height = height;
    }

    public float getLength() {
        return length;
    }

    public float getWidth() {
        return width;
    }

    public float getHeight() {
        return height;
    }

    protected abstract boolean validate();
}

File: PostBox/src/main/kotlin/com/mvp/bean/BaseBox.kt
package com.mvp.bean.kt

abstract class BaseBox(val length: Float?,
                       val width: Float?,
                       val height: Float?) {

    val lengthOfBox3 = 23f
    val widthOfBox3 = 14f
    val heightOfBox3 = 13f

    val lengthOfBox5 = 39.5f
    val widthOfBox5 = 27.5f
    val heightOfBox5 = 23f

    fun validateReceivedBox(constraintLength: Float?,
                            constraintWidth: Float?,
                            constraintHeight: Float?) : Boolean {

        val totalConstraintSize = constraintLength!! + constraintWidth!! + constraintHeight!!
        val totalInputSize = length!! + width!! + height!!

        if (length!! > constraintLength || width!! > constraintWidth || height!! > constraintHeight)
            return false
        else if (totalInputSize <= totalConstraintSize)
            return true
        else
            return true

    }
}